alert-service/service/service_test.go

```diff
@@ -114,6 +204,128 @@ func TestProcessHostDataInsertion_DiffHostError3(t *testing.T) {
+		"MAJOR This is just an alert test to a mocked emailer. on TestHostname",
+		`Date: 2019-09-02 10:25:28 +0000 UTC
+Severity: MAJOR
+Host: TestHostname
+Code: NEW_LICENSE
+This is just an alert test to a mocked emailer.`,
+		as.Config.AlertService.Emailer.To)
+
+	params := make(hub.Fields, 1)
+	params["alert"] = model.Alert{
+		OtherInfo:     map[string]interface{}{"Hostname": "TestHostname"},
+		AlertSeverity: model.AlertSeverityMajor,
+		Description:   "This is just an alert test to a mocked emailer.",
+		Date:          utils.P("2019-09-02T10:25:28Z"),
+		AlertCode:     model.AlertCodeNewLicense,
+	}
+
+	as.ProcessAlertInsertion(params)
+}
+
+func TestProcessAlertInsertion_WithoutHostname(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	emailer := NewMockEmailer(mockCtrl)
+
+	as := AlertService{
+		Emailer: emailer,
+		TimeNow: utils.Btc(utils.P("2019-11-05T16:02:03Z")),
+		Log:     utils.NewLogger("TEST"),
+		Queue:   hub.New(),
+		Config: config.Configuration{
+			AlertService: config.AlertService{
+				Emailer: config.Emailer{
+					To: []string{"test@ercole.test"},
+				},
+			},
+		},
+	}
+
+	emailer.EXPECT().SendEmail(
+		"MAJOR This is just an alert test to a mocked emailer.",
+		`Date: 2019-09-02 10:25:28 +0000 UTC
+Severity: MAJOR
+Code: NEW_LICENSE
+This is just an alert test to a mocked emailer.`,
+		as.Config.AlertService.Emailer.To)
+
+	params := make(hub.Fields, 1)
+	params["alert"] = model.Alert{
+		OtherInfo:     map[string]interface{}{},
+		AlertSeverity: model.AlertSeverityMajor,
+		Description:   "This is just an alert test to a mocked emailer.",
+		Date:          utils.P("2019-09-02T10:25:28Z"),
+		AlertCode:     model.AlertCodeNewLicense,
+	}
+
+	as.ProcessAlertInsertion(params)
+}
+
+func TestProcessAlertInsertion_EmailerError(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	emailer := NewMockEmailer(mockCtrl)
+
+	as := AlertService{
+		Emailer: emailer,
+		TimeNow: utils.Btc(utils.P("2019-11-05T16:02:03Z")),
+		Log:     utils.NewLogger("TEST"),
+		Queue:   hub.New(),
+		Config: config.Configuration{
+			AlertService: config.AlertService{
+				Emailer: config.Emailer{
+					To: []string{"test@ercole.test"},
+				},
+			},
+		},
+	}
+
+	emailer.EXPECT().SendEmail(
+		"MAJOR This is just an alert test to a mocked emailer.",
+		`Date: 2019-09-02 10:25:28 +0000 UTC
+Severity: MAJOR
+Code: NEW_LICENSE
+This is just an alert test to a mocked emailer.`,
+		as.Config.AlertService.Emailer.To).
+		Return(utils.NewAdvancedErrorPtr(fmt.Errorf("test error from emailer"), "test EMAILER"))
+
+	params := make(hub.Fields, 1)
+	params["alert"] = model.Alert{
+		OtherInfo:     map[string]interface{}{},
+		AlertSeverity: model.AlertSeverityMajor,
+		Description:   "This is just an alert test to a mocked emailer.",
+		Date:          utils.P("2019-09-02T10:25:28Z"),
+		AlertCode:     model.AlertCodeNewLicense,
+	}
+
+	as.ProcessAlertInsertion(params)
+}
```

Comment:
What are you verifying?

Reply:
I'm verifying that even if the emailer behave wrongly and return an error, this error is managed.